src/main/kotlin/build/buf/intellij/index/ModuleKeyIndex.kt

```diff
@@ -106,20 +97,24 @@ class BufModuleKeyIndex : ScalarIndexExtension<ModuleKey>() {
             return null
         }
     }
+
+    companion object {
+        private val LOG = logger<ModuleKeyIndex>()
+    }
 }
 
-object BufModuleIndexEntryKeyDescriptor : KeyDescriptor<ModuleKey> {
-    override fun getHashCode(value: ModuleKey): Int = value.hashCode()
+object ModuleKeyIndexEntryKeyDescriptor : KeyDescriptor<ModuleKey> {
+    override fun getHashCode(value: ModuleKey): Int = Objects.hash(value.moduleFullName, value.commitID)
 
-    override fun isEqual(a: ModuleKey, b: ModuleKey): Boolean = a == b
+    override fun isEqual(a: ModuleKey, b: ModuleKey): Boolean = a.moduleFullName == b.moduleFullName && a.commitID == b.commitID
```

Comment:
This seems like a strange place for this. Why not push this down to `ModuleKey` instead?

Reply:
Reverted this change - I was playing around with using a standard data class for ModuleKey and customizing it for index usage here, but ended up not making that change. Reverted.